### PR TITLE
Update ConstRefVector.h

### PR DESCRIPTION
**Pull request by timosachsenberg**

### CODE REVIEW

**Comment by timosachsenberg** [comment]
Is it defined twice then? If so the second one should be completely removed

File: src/openms/include/OpenMS/DATASTRUCTURES/ConstRefVector.h [hunk 1]
```diff
@@ -214,7 +214,7 @@ namespace OpenMS
         return tmp;
       }
 
-      reference operator*()
+      reference operator*() const
```
Comment (hroest) [inline src/openms/include/OpenMS/DATASTRUCTURES/ConstRefVector.h]: why do we have 

```
pointer operator->();
pointer operator->() const;
```

but only

```
reference operator*() const;
```

I am not sure that this makes sense, actually it is unclear why we have the same function as const and non-const, there isnt really a great usecase for the non-const function, is there?

Comment (timosachsenberg) [reply src/openms/include/OpenMS/DATASTRUCTURES/ConstRefVector.h]: Doesn’t make sense... is it used anywhere?

Comment (timosachsenberg) [reply src/openms/include/OpenMS/DATASTRUCTURES/ConstRefVector.h]: I removed them... did still compile

**Pull request merged**